### DISCUSSION TITLE
Dimmer esp8266

### DISCUSSION THREAD

**Discussion by denismsd** [arendst/Tasmota#16043] [Q&A] [answered]
hello guys I bought a dimmer board for tasmota on esp8266 but I can't make it work, I didn't find anything about it, the board has vcc, zc, dim, gnd I followed the tasmota guide steps but it doesn't work
![IMG_20220721_151726](https://user-images.githubusercontent.com/76956475/180286564-ae9194a0-c0f0-44c0-8da3-9efad264a499.jpg)

**Comment by Gongman514** [comment]
Who makes the board? Looks interesting.

Cheers

**Reply by denismsd** [reply]
here in Brazil there is a website that sells it.
https://www.usinainfo.com.br/dimmer-arduino/modulo-dimmer-para-arduino-pic-mc-8a-com-sinal-zero-cross-220v-5915.html

**Comment by sfromis** [comment, marked as answer]
While this is not the same board, seeing ZC (zerocross) I suspect that it may be "similar" to the Robotdyn AC Dimmer board referred to here:
https://tasmota.github.io/docs/Lights/#1-channel-dimmable-light

**Reply by denismsd** [reply]
I changed it to esp32 it worked, but sometimes the lamp that is on the load goes out then comes back on its own, it must be the ZC and PWM function configuration but I don't know how to solve it

**Reply by sfromis** [reply]
In general, when things do not work out, do check the log (console or serial) at log level 3 or 4, for details of Tasmota actions.

**Reply by denismsd** [reply]
hello guys all right, I ended up testing it again with esp32, I think the problem is in the synchronization with ZC, in setoption99 it talks about this synchronization (but I don't know which command changes it) I changed the pemfrequecy function to 60HZ.  it starts working normally then it starts to dim then it goes back to normal.

can you tell if there is any other configuration to synchronize zerocross and pwm

SetOption99 Enable zero-cross capable AC dimmer
0 = no zero-cross AC dimmer connected (default)
1 = zero-cross AC dimmer attached. Focus on raising edge and sync frequency

**Reply by cacengineering** [reply]
I am having the same problem on an ESP32.  Use an oscilloscope to monitor both the mains in and the PWM - the pwm, despite having setoption99 1 set, drifts.  I have however set my mains frequency as 100Hz as we are using a rectifier before optocoupler - so mains ZC does look like 100Hz.  Am wondering if the software adjusts for this and might try 50Hz.  Not that hopeful.

Scope screenshot below - top line is rectified mains (able to trigger counter) and bottom is output of pwm.  They just slowly drift in relation to each other.

![image](https://github.com/arendst/Tasmota/assets/110654750/38c43e15-f1df-4459-859f-8bc3371b84a1)
